File: Computing/C_Practice/revision/BOON_LAY/T4_SIMPLENUM_20_Nikola.c
#include <stdio.h>
#include <stdlib.h>

float average(int numbers[]);
void user_input();

int main() {

    return 0;
}

void user_input() {

    char ch;
    int n;
    int numbers[n];
    int index = 0;

    printf("Welcome to Simple Numbers\n");
    printf("Numbers please.\n");
    printf("Separate each number using spaces: ");
    while (ch = getchar()) {
        if (ch == ' ') {
            continue;
        } else if (ch == ',') {
            continue;
        } else {
            numbers[index] = atoi(ch);
        }
    }
}

float average(int numbers[]) {
    int *p;
    int sum = 0;
    int numbers_length = sizeof(numbers) / sizeof(numbers[0]);

    for (p = numbers; p < numbers_length; p++) {
        sum += *p;
    }

    return sum / numbers_length;
}

File: Computing/MTR/2022/Boon_Lay/T4_SIMPLENUM_20_Nikola.py
def average(num_list):
    return sum(num_list) / len(num_list)

def user_input():
    print("Welcome to Simple Numbers")
    print("Numbers please.")
    numbers = []
    check = False

    while True:
        numbers = input("Separate each number using spaces: ").split(" ")

        for i in numbers:
            if not(i.isalnum()):
                print("Inputs have to be valid digits.")
                check = True
                break

        if check:
            pass
        else:
            break
            

    return [int(k) for k in numbers]

def main():
    numbers = user_input()
    print("Average: {}".format(average(numbers)))
    print("Maximum: {}".format(max(numbers)))
    print("Minimum: {}".format(min(numbers)))

main()
